code-challenges.test.js

```diff
@@ -23,32 +30,122 @@ var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saf
 
 
 // b) Create the function that makes the test pass.
-
-
-
+/*
+   Algorithm->remove first element of an array and shuffle randomly the remaining element of the array
+        Input->an array
+        Output->randomly shuffeld array and 1 less than array.length
+            create a function
+            remove the first element of the array using shift() built in method
+            loop the array from array.length to array >zero
+            declare variable j,use Math.floor and Math.random built in methods
+            swap i and j
+            return array
+*/
+const shuffle = array => {
+    array.shift()
+    for (let i = array.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i],array[j]]=[array[j],array[i]]
+    }return array
+  }
+  
```

Comment:
I am wondering how often you were able to get this test to pass? Logic does look good though!